fud/fud/exec.py

```diff
@@ -131,37 +131,24 @@ def run_fud(args, config):
             # Whether this should output data in CSV format.
             is_csv = any(a == "csv" for a in profiled_stages)
             # If no stages provided, print overall profiling information for the stages.
-            if not profiled_stages or (len(profiled_stages) == 1 and is_csv):
-                kwargs = {
-                    "stage": "stage",
-                    "phases": [ed for ed in path],
-                    "durations": overall_durations,
-                }
-                data.data = (
-                    utils.profiling_csv(**kwargs)
-                    if is_csv
-                    else utils.profiling_dump(**kwargs)
+            if profiled_stages in [[], ["csv"]]:
+                data.data = utils.profile_stages(
+                    "stage", [ed for ed in path], overall_durations, is_csv
                 )
             else:
                 # Otherwise, gather profiling data for each stage provided.
                 def gather_profiling_data(stage):
-                    if stage == "csv":
-                        return ""
-                    if stage not in collected_profiling_stages:
+                    if stage not in collected_for_profiling:
                         raise errors.UndefinedStage(stage)
-                    kwargs = {
-                        "stage": stage,
-                        "phases": [s for s in collected_profiling_stages[stage].steps],
-                        "durations": collected_profiling_stages[stage].durations,
-                    }
-                    return (
-                        utils.profiling_csv(**kwargs)
-                        if is_csv
-                        else utils.profiling_dump(**kwargs)
+                    return utils.profile_stages(
+                        stage,
+                        [s for s in collected_for_profiling[stage].steps],
+                        collected_for_profiling[stage].durations,
+                        is_csv,
                     )
 
                 data.data = "\n".join(
-                    gather_profiling_data(stage) for stage in profiled_stages
+                    gather_profiling_data(s) for s in profiled_stages if s != "csv"
```

Comment:
Is there a way to not hardcode the name of the stage here?

Reply:
The flag takes in a list of stages and `csv` if we want CSV format. Are you saying pass in `csv` in a separate flag or something?

Reply:
Maybe I misread this–it seems that there is a stage named "csv" that is being ignored in this statement.

Reply:
Hm, that's a little odd. It seems `csv` should be flag, not a stage because it is not really transforming the input states in any way. It is instead collecting information about the execution. If this is a simple change to make, make csv a flag. If not, open an issue and leave it be for after the deadline.

Reply:
Are you saying something like:

```
--dump_prof futil --some_csv_flag
```

?

Reply:
Yeah

Reply:
Cool. Good idea. I'll knock that after work in a separate CL.